spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindOverridableMethodCall.java

```diff
@@ -0,0 +1,274 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package edu.umd.cs.findbugs.detect;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.JavaClass;
+
+import edu.umd.cs.findbugs.BugAccumulator;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.OpcodeStack;
+import edu.umd.cs.findbugs.SourceLineAnnotation;
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import edu.umd.cs.findbugs.ba.XClass;
+import edu.umd.cs.findbugs.ba.XMethod;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import edu.umd.cs.findbugs.classfile.CheckedAnalysisException;
+import edu.umd.cs.findbugs.classfile.ClassDescriptor;
+import edu.umd.cs.findbugs.util.MultiMap;
+
+public class FindOverridableMethodCall extends OpcodeStackDetector {
+
+    private static class CallerInfo {
+        XMethod method;
+        SourceLineAnnotation sourceLine;
+
+        CallerInfo(XMethod m, SourceLineAnnotation sl) {
+            method = m;
+            sourceLine = sl;
+        }
+    }
+
+    private static final Map<XMethod, CallerInfo> callerConstructors = new HashMap<>();
+    private static final Map<XMethod, CallerInfo> callerClones = new HashMap<>();
+    private static final Map<XMethod, XMethod> callsToOverridable = new HashMap<>();
+    private static final MultiMap<XMethod, XMethod> callerToCalleeMap = new MultiMap<>(ArrayList.class);
+    private static final MultiMap<XMethod, XMethod> calleeToCallerMap = new MultiMap<>(ArrayList.class);
+
+
+    private final BugAccumulator bugAccumulator;
+
+    public FindOverridableMethodCall(BugReporter bugReporter) {
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visit(JavaClass obj) {
+        super.visit(obj);
+        callerConstructors.clear();
+        callerClones.clear();
+        callsToOverridable.clear();
+        callerToCalleeMap.clear();
+        calleeToCallerMap.clear();
+    }
+
+    @Override
+    public void visitAfter(JavaClass obj) {
+        bugAccumulator.reportAccumulatedBugs();
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
+        if (seen == Const.INVOKEINTERFACE || seen == Const.INVOKEVIRTUAL) {
```

Comment:
It could be better to consider the method references like `optional.orElseGet(this::generate)`, but it's a minor topic because most of existing detectors do not cover it either.